client/packageutils.c

```diff
@@ -1148,6 +1148,19 @@ TDNFPopulatePkgInfos(
                       &pPkgInfo->pszLocation);
         BAIL_ON_TDNF_ERROR(dwError);
 
+        dwError = SolvGetPkgChecksumFromId(
+                      pSack,
+                      dwPkgId,
+                      REPOKEY_TYPE_SHA256,
+                      SIZEOF_SHA256,
```

Comment:
This function only support for SHA256 ?

Reply:
> This function only support for SHA256 ?

If this is not specific for SHA256  then we can consider to check other checksum as well.
But if the expectation is only for SHA256 then its fine.

Reply:
I commented before that repodata can have other checksum types.